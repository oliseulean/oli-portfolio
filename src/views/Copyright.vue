<template>
  <div
    class="app__footer"
    v-html="
      `&copy; Copyright &copy; ${getCurrentYear()} | All rights reserved. <br /> Designed by Olimpiu È˜eulean`
    "
  />
</template>

<script setup>
/*
 * Imports
 */
import getCurrentYear from '../helpers/getCurrentYear';
</script>

<style lang="scss" scoped>
.app__footer {
  @include font-roboto-slab();
  color: $color-light;
  background: $color-dark-blue;
  bottom: 0;
  left: 0;
  width: 100%;
  text-align: center;
  font-size: 0.8rem;
  display: block;
  margin: 0 auto;
  padding: 1rem;
}
</style>
