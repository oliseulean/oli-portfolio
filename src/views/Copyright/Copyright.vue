<script setup>
/* Imports */
import getCurrentYear from '../../helpers/getCurrentYear';
</script>

<template>
  <div
    class="app-footer"
    v-html="
      `&copy; Copyright &copy; ${getCurrentYear()} | All rights reserved.
      <br />
      Designed by Olimpiu È˜eulean`
    "
  />
</template>

<style lang="scss" scoped>
.app-footer {
  display: block;
  margin: 0 auto;

  @include font-roboto-slab();

  color: $color-catskill-white;
  background: $color-black-pearl;
  bottom: 0;
  left: 0;
  width: 100%;
  text-align: center;
  font-size: 0.8rem;
  padding: 1rem;
}
</style>
