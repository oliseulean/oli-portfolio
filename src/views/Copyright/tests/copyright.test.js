import { mount } from '@vue/test-utils';
import Copyright from '../../Copyright/Copyright.vue';
import { it, describe, expect, beforeEach } from 'vitest';

describe('Copyright', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(Copyright);
  });

  it('component renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render the author name', () => {
    expect(wrapper.html()).toContain('Designed by Olimpiu È˜eulean');
  });
});
