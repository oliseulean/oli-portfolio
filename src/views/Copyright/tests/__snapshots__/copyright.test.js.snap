// Vitest Snapshot v1

exports[`Copyright > component renders correctly 1`] = `
"<div class=\\"app-footer\\" data-v-591b5195=\\"\\">© Copyright © 2023 | All rights reserved.
  <br>
  Designed by Olimpiu Șeulean
</div>"
`;
